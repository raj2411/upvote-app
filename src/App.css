.App {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  flex-direction: column;
  background-color: #f4f4f4;
  margin: 0;
}

h1 {
  margin-bottom: 20px;
}

.upvote-lists-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
}
