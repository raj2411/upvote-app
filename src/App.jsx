import React from 'react';
import './App.css';
import UpvoteList from './components/upVoteList';

function App() {
  return (
    <div className="App">
      <h1>Upvote Lists</h1>
      <div className="upvote-lists-container">
        <UpvoteList storageKey="upvoteList1" />
        <UpvoteList storageKey="upvoteList2" />
        <UpvoteList storageKey="upvoteList3" />

      </div>
    </div>
  );
}

export default App;