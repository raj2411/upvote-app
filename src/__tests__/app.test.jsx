import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import App from '../App';
import '@testing-library/jest-dom'; 

describe('App component', () => {
  it('should render the App component with three UpvoteLists', () => {
    render(<App />);
    
    const titleElement = screen.getByText(/Upvote Lists/i);
    expect(titleElement).toBeInTheDocument(); 
    
    const upvoteLists = screen.getAllByTestId('upvote-list');
    expect(upvoteLists.length).toBe(3);
  });
});
