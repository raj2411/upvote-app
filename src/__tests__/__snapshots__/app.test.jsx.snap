// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Upvote Lists
    </h1>
    <div
      class="upvote-lists-container"
    >
      <div
        class="upvote-list"
      >
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <button
          class="add-button"
          data-testid="add-button"
        >
          +
        </button>
      </div>
      <div
        class="upvote-list"
      >
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <button
          class="add-button"
          data-testid="add-button"
        >
          +
        </button>
      </div>
      <div
        class="upvote-list"
      >
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <div
          class="upvote "
          data-testid="upvote-button"
        >
          <svg
            class="feather feather-arrow-up"
            fill="none"
            height="24"
            stroke="#343A40"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </div>
        <button
          class="add-button"
          data-testid="add-button"
        >
          +
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
